cmd/snap/cmd_warnings.go

```diff
@@ -0,0 +1,210 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/jessevdk/go-flags"
+
+	"github.com/snapcore/snapd/client"
+	"github.com/snapcore/snapd/i18n"
+	"github.com/snapcore/snapd/osutil"
+)
+
+type cmdWarnings struct {
+	clientMixin
+	timeMixin
+	All     bool `long:"all"`
+	Verbose bool `long:"verbose"`
+}
+
+type cmdOkay struct{ clientMixin }
+
+var shortWarningsHelp = i18n.G("List warnings")
+var longWarningsHelp = i18n.G(`
+The warnings command lists the warnings that have been reported to the system.
+
+Once warnings have been listed with 'snap warnings', 'snap okay' may be used to
+silence them. A warning that's been silenced in this way will not be listed
+again unless it happens again, _and_ a cooldown time has passed.
+
+Warnings expire automatically, and once expired they are forgotten.
+`)
+
+var shortOkayHelp = i18n.G("Acknowledge warnings")
+var longOkayHelp = i18n.G(`
+The okay command acknowledges the warnings listed with 'snap warnings'.
+
+Once acknowledged a warning won't appear again unless it re-occurrs and
+sufficient time has passed.
+`)
+
+func init() {
+	addCommand("warnings", shortWarningsHelp, longWarningsHelp, func() flags.Commander { return &cmdWarnings{} }, timeDescs.also(map[string]string{
+		"all":     i18n.G("Show all warnings"),
+		"verbose": i18n.G("Show more information"),
+	}), nil)
+	addCommand("okay", shortOkayHelp, longOkayHelp, func() flags.Commander { return &cmdOkay{} }, nil, nil)
+}
+
+func (cmd *cmdWarnings) Execute(args []string) error {
```

Comment:
This is also not tested, adding a test would be great. Maybe it will result in some small refactor to extract bits from execute (but maybe not).

Reply:
it's covered now